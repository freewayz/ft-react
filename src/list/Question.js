import React from 'react'
import {Link} from 'react-router-dom'

const Question = ({question}) => {
  const {title, id} = question
  const editLink = `/edit/${id}`
  return (
    <div className='question__item'>
      <div> 8 </div>
      <div> Recieved </div>
      <div>
        <a href={editLink}>
          {title}
        </a>
      </div>
      <div> Libarary</div>
      <div>Published </div>
      <div> Multichoice </div>
      <div> XML </div>
    </div>

  )
}

export default Question
