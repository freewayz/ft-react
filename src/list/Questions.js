import React, {Component} from 'react'
import Question from './Question'

class Questions extends Component {
  render () {
    const {questions} = this.props

    return (
      <div>
        {
          questions.map((question, index) => {
            return <Question {...question} key={index} />
          })
        }
      </div>
    )
  }
}

export default Questions
