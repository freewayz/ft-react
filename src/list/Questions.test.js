/* global expect jest it describe shallow */
import React from 'react'
import crateRouterContext from 'react-router-test-context'
import Questions from './Questions'
import Question from './Question'

const questionProps = {
  title: 'Question Title',
  description: 'Question Description',
  id: 3
}

const questions = []
for (let i = 0; i < 10; i++) {
  questions.push(questionProps)
}

const context = crateRouterContext()
describe('Questions Component Testcast', () => {
  it('create the right snapshots', () => {
    const tree = shallow(<Questions questions={questions} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders 10 Question functional component', () => {
    const wrapper = shallow(<Questions questions={questions} />, {context})
    const QuestionChildren = wrapper.find(Question)
    expect(QuestionChildren.length).toBe(10)
  })

  it('renders no Question component for empty array', () => {
    const wrapper = shallow(<Questions questions={[]} />, {context})
    const QuestionChildren = wrapper.find(Question)
    expect(QuestionChildren.length).toBe(0)
  })
})
