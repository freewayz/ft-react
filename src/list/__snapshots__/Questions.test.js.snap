// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions Component Testcast create the right snapshots 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Questions
    questions={
        Array [
            Object {
              "description": "Question Description",
              "id": 3,
              "title": "Question Title",
            },
            Object {
              "description": "Question Description",
              "id": 3,
              "title": "Question Title",
            },
            Object {
              "description": "Question Description",
              "id": 3,
              "title": "Question Title",
            },
            Object {
              "description": "Question Description",
              "id": 3,
              "title": "Question Title",
            },
            Object {
              "description": "Question Description",
              "id": 3,
              "title": "Question Title",
            },
            Object {
              "description": "Question Description",
              "id": 3,
              "title": "Question Title",
            },
            Object {
              "description": "Question Description",
              "id": 3,
              "title": "Question Title",
            },
            Object {
              "description": "Question Description",
              "id": 3,
              "title": "Question Title",
            },
            Object {
              "description": "Question Description",
              "id": 3,
              "title": "Question Title",
            },
            Object {
              "description": "Question Description",
              "id": 3,
              "title": "Question Title",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Question
          description="Question Description"
          id={3}
          title="Question Title"
/>,
        <Question
          description="Question Description"
          id={3}
          title="Question Title"
/>,
        <Question
          description="Question Description"
          id={3}
          title="Question Title"
/>,
        <Question
          description="Question Description"
          id={3}
          title="Question Title"
/>,
        <Question
          description="Question Description"
          id={3}
          title="Question Title"
/>,
        <Question
          description="Question Description"
          id={3}
          title="Question Title"
/>,
        <Question
          description="Question Description"
          id={3}
          title="Question Title"
/>,
        <Question
          description="Question Description"
          id={3}
          title="Question Title"
/>,
        <Question
          description="Question Description"
          id={3}
          title="Question Title"
/>,
        <Question
          description="Question Description"
          id={3}
          title="Question Title"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "description": "Question Description",
          "id": 3,
          "title": "Question Title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "description": "Question Description",
          "id": 3,
          "title": "Question Title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "description": "Question Description",
          "id": 3,
          "title": "Question Title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "description": "Question Description",
          "id": 3,
          "title": "Question Title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "description": "Question Description",
          "id": 3,
          "title": "Question Title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "function",
        "props": Object {
          "description": "Question Description",
          "id": 3,
          "title": "Question Title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "function",
        "props": Object {
          "description": "Question Description",
          "id": 3,
          "title": "Question Title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "function",
        "props": Object {
          "description": "Question Description",
          "id": 3,
          "title": "Question Title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "8",
        "nodeType": "function",
        "props": Object {
          "description": "Question Description",
          "id": 3,
          "title": "Question Title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "9",
        "nodeType": "function",
        "props": Object {
          "description": "Question Description",
          "id": 3,
          "title": "Question Title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Question
            description="Question Description"
            id={3}
            title="Question Title"
/>,
          <Question
            description="Question Description"
            id={3}
            title="Question Title"
/>,
          <Question
            description="Question Description"
            id={3}
            title="Question Title"
/>,
          <Question
            description="Question Description"
            id={3}
            title="Question Title"
/>,
          <Question
            description="Question Description"
            id={3}
            title="Question Title"
/>,
          <Question
            description="Question Description"
            id={3}
            title="Question Title"
/>,
          <Question
            description="Question Description"
            id={3}
            title="Question Title"
/>,
          <Question
            description="Question Description"
            id={3}
            title="Question Title"
/>,
          <Question
            description="Question Description"
            id={3}
            title="Question Title"
/>,
          <Question
            description="Question Description"
            id={3}
            title="Question Title"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "description": "Question Description",
            "id": 3,
            "title": "Question Title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "description": "Question Description",
            "id": 3,
            "title": "Question Title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "description": "Question Description",
            "id": 3,
            "title": "Question Title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "description": "Question Description",
            "id": 3,
            "title": "Question Title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "description": "Question Description",
            "id": 3,
            "title": "Question Title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "description": "Question Description",
            "id": 3,
            "title": "Question Title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "function",
          "props": Object {
            "description": "Question Description",
            "id": 3,
            "title": "Question Title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "function",
          "props": Object {
            "description": "Question Description",
            "id": 3,
            "title": "Question Title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "function",
          "props": Object {
            "description": "Question Description",
            "id": 3,
            "title": "Question Title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "function",
          "props": Object {
            "description": "Question Description",
            "id": 3,
            "title": "Question Title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
